src/ui/Code.ts

```diff
@@ -142,3 +142,18 @@ function CHECK_ECDSA_SIGNATURE(
 function RANGE_CHECK(num: number): bigint {
   return range_check(BigInt(num));
 }
+
+function KECCAK(message: number | string): number | string {
+  const utf8Bytes = encodeUTF8(message.toString());
+  const bytearrayOutput = Keccak(1088, 512, utf8Bytes, 0x01, 256 / 8);
+
+  const bitOutput = Array.from(bytearrayOutput)
+    .map((byte) => byte.toString(2).padStart(8, "0"))
+    .join("");
+
+  const bitStringSlice = bitOutput.slice(6);
+  const bitBigInt = BigInt("0b" + bitStringSlice);
+
+  const hexOutput = "0x" + bitBigInt.toString(16);
+  return hexOutput;
```

Comment:
I don't understand this

Reply:
Starknet keccak is the 250 first bits of the regular keccak-256, so this converts keccak output to bits, selects the 250 first and converts it to hex